digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadNetworkDAO.scala

```diff
@@ -122,15 +122,27 @@ class RoadNetworkDAO {
     * Fetches the data required for /summary API result generation.
     * @return The road address information, and road names of the whole latest road network
     */
-  def fetchRoadwayNetworkSummary: Seq[RoadwayNetworkSummaryRow] = {
+  def fetchRoadwayNetworkSummary(date: Option[DateTime] = None): Seq[RoadwayNetworkSummaryRow] = {
     time(logger, "Get whole network summary") {
+      val (dateFilter, roadnameTable) = date match {
+        case Some(date) =>
+          val dateString = date.toString("yyyy-MM-dd")
+          val datefilter =  s"r.START_DATE <= to_date('$dateString}', 'yyyy-MM-dd') " +
+          s"AND (r.END_DATE IS NULL OR to_date('$dateString', 'yyyy-MM-dd') <= r.END_DATE) "
+
+          val roadnamefilter = s"(SELECT * FROM ROAD_NAME rn WHERE rn.START_DATE <= to_date('$dateString', 'yyyy-MM-dd')" +
+            s"AND (rn.END_DATE IS NULL OR to_date('$dateString', 'yyyy-MM-dd') <= rn.END_DATE) and rn.VALID_TO IS NULL)"
+          (datefilter, roadnamefilter)
+        case None => ("r.END_DATE IS NULL", "(SELECT * FROM ROAD_NAME rn WHERE rn.END_DATE IS NULL)")
```

Comment:
Road_name table contains multiple entries for roadnumbers so there is a need to ensure that only one road_name table entry is selected for each roadnumber in order to avoid duplicates in the roadway network summary.